### ISSUE TITLE
USB3 bus lockup when using CueRipper 2.15-7

### ISSUE THREAD

**Issue by itrushton** [gchudov/cuetools.net#71]
I'm using a Lindy USB3 5.25 extyernal enclosure with various Liteon/Pioneer drives installed.  when connecting the enclosure, it's detected and reads the disc tracks and collects metadata, but as soon as i ask it to Rip it gets to "detecting drive features" and the drive access light flickers and thats it.  sometimes after a minute i get an IoCtrlError message, sometimes not; but alsways it results in a hung CueRipper window where both pause/abort are greyed out; then i start getting odd USB and machine issues (lockups of other devices, OS freezing with spinning circle etc) - until i either software reset the port, or unplug and reinsert. If i connect the device to a USB 2.1 port it works perfectly.  I have other USB3 external enclosures using prolific chipsets that work every time; only these ones (which i believe use the VIA chipset).  Clean install on Windows 10 (full format, not just reinstall). zero errors in windows event logs

**Comment by itrushton** [comment]
I've now tried a number of different enclosures, and noen of them exhibit this same issue wit CueRipper - sadly the ones having the issue are the decent ones!  I have even managed to locate a firmware update for the controller and applied it - but no difference at all.  Would really like to use my external drives with CueRipper as none of the others are as good.  happy to provide logs/carry out testing if it will help?  The chipset in the enclosure is the VIA VL700-Q4 - it must be a specific request that causes the problem.  the drive spins up, identifies the disc, pulls the info down, i click go, it says "detecting drive features" - i hear the disc start to spin up and the light might flash briefly - then it stops.  after a while i might get a report saying "unable to read IOctl error", other times it just hangs.  but after this happens, the USB bus is completely erratic until i disconnect the enclosure nd restart it.

**Comment by itrushton** [comment]
I've run a bus analyser on a good enclosure and the Lindy enclosure to try and establish what's going on - images of bus traffic attached.  interesting to note that when the lindy enclosure fails it does literally stop all bus traffic.  I've also now tried the same process using a Z97 motherboard with native USB3.0 - this has exactly the same issues.  Would be really good if someone could help with this - I really don't want to have to return these enclosures as they are the only decent ones i've found!  Unfortunately I can't find any logs for CueRipper anywhere to attach to try and track down the issue myself.  Can someone assist please?
<img width="1364" alt="failed Lindy enclosure rip" src="https://user-images.githubusercontent.com/9013765/107131908-c6a42880-68d2-11eb-9bb5-f34430525472.png">
<img width="1359" alt="OWC working" src="https://user-images.githubusercontent.com/9013765/107131916-cdcb3680-68d2-11eb-8de0-766317b5cf51.png">

**Comment by c72578** [comment]
@itrushton Thanks for this report and the investigations.
So, in summary the drives are working properly, when they are connected using their native interface (e.g. SATA). Different USB enclosures have been tested. All of them work on USB 2.1 ports. Several of them also work on USB 3 ports.
However, there are Lindy USB3 5.25 external enclosures, which only work properly on USB 2.1 ports and cause problems using CUERipper, when they are connected to USB 3 ports. The chipset of these Lindy USB3 5.25 external enclosures is VIA VL700-Q4.

Could you please also share a product number and a product link of the affected Lindy USB3 5.25 external enclosures?
Thanks

**Comment by itrushton** [comment]
Hi Wolfgang
Thank you for getting back to me. Please see the link below:
USB 3.0 Enclosure 5.25" for BD/DVD/CD Drives - from LINDY UK 

  
|  
|   
|   
|   £ 60.66   |    |

   |

  |
|  
|   |  
USB 3.0 Enclosure 5.25" for BD/DVD/CD Drives - from LINDY UK
 
USB 3.0 Enclosure 5.25" for BD/DVD/CD Drives USB Black, USB 3.0, BD/DVD/CD Drive, USB 3.0, USB 3.0 B Female to SATA 22 Pin Male @ LINDY. The UKs leading HDMI Cable, USB Cable and FireWire Cable supplier. Also supplies connectors, switches and ad
  |   |

  |

  |

  
Trying to force onto USB 2.1 is a little difficult due to the decreasing number of usb2 ports now on machines - ironically I bought these enclosures because some older usb2.0 ones I had would not work on a newer machine!
Many thanks


Sent from Yahoo Mail for iPhone


On Tuesday, February 9, 2021, 12:58 pm, Wolfgang Stöggl <notifications@github.com> wrote:




@itrushton Thanks for this report and the investigations.
So, in summary the drives are working properly, when they are connected using their native interface (e.g. SATA). Different USB enclosures have been tested. All of them work on USB 2.1 ports. Several of them also work on USB 3 ports.
However, there are Lindy USB3 5.25 external enclosures, which only work properly on USB 2.1 ports and cause problems using CUERipper, when they are connected to USB 3 ports. The chipset of these Lindy USB3 5.25 external enclosures is VIA VL700-Q4.

Could you please also share a product number and a product link of the affected Lindy USB3 5.25 external enclosures?
Thanks

**Comment by c72578** [comment]
OK, thanks. It is a Lindy USB 3.0 Enclosure 5.25" for BD/DVD/CD Drives
No. 43138
EAN: 4002888431385

**Comment by itrushton** [comment]
Sorry - I hadn't realised the link wouldn't populate correctly!

**Comment by c72578** [comment]
@itrushton Have you already tried a different USB cable (e.g. USB 2.0 extension cable, see below)?
See (VIA VL700-Q4 related):
https://answers.microsoft.com/en-us/windows/forum/windows_7-hardware/usb3-dvd-brief-data-flow/0c207662-012d-4952-8ff8-cd6b0e5df050

From https://www.newegg.com/bytecc-me-535u3/p/N82E16817145090?Item=N82E16817145090&Tpk=ME-535U3
e.g.:
> Overall Review: My DVD-RW drive would drop out after about 5 minutes or so of use while using the included usb 3.0 cable. Don't think this is a problem with the enclosure as much as it is with windows having issues with a DVD drive over usb 3.0. I've noticed similar issues across several forums. Haven't tried with a bluray drive or hard drive, so can't give any feedback on that. Just be aware that you will need to use a 2.0 cable if using the enclosure with a DVD drive.

or:

>  In connecting it in the USB 3.0 port, the Bluray player was not detected by Windows. In pluging it in a USB 2.0 extension cable and in pluging this one to the USB 3.0 port of my notebook, it works very well.

**Comment by itrushton** [comment]
@c72578 thanks for the links.  I had seen the first article - but only just realised they are one and the same enclosure - the BYTECC one being significantly cheaper than the Lindy variation!

I at first thought it was a suspect cable, but it was in fact plugging the drive into a USB3 port when powered up that causes it to only connect at 2.1 on a USB port (most of the time), which then causes it to work.    I have since tried multiple different USB3 cables (including ones which work perfectly with other enclosures) - and have even installed ferrite Chokes on cables to ensure there isn't any interference being cause.  I've put them in different locations, using different power supplies in different sockets, but always the same result - identifies the disk, then crashes and takes the whole USB bus with it (as well as the machine). I even resorted to flashing a newer firmware from a different enclosure using the same chipset - also with no difference.

While I haven't tried burning a disc, I have ripped several audio CDs in secure mode on EAC (current version) without any issues, and also used DBpoweramp to rip a couple of discs successfully; so it does work ok on USB3, it just seems to be something specific that Cueripper does it really doesn't like.  Like I say, more than happy to run any sort of diagnostics or logging to provide you if it will help!

**Comment by c72578** [comment]
@itrushton The trick is to use a USB 2.0 cable.

**Comment by itrushton** [comment]
Ahh ok I see what you mean!  
So what is causing this problem with CueRipper that causes it to crash the bus when other ripping tools like EAC work without issue?  does EAC handle things differently in some way?

**Comment by bleggett** [comment]
Coming over from #164 with a BU40N and a USB3 enclosure - 

Confirming that using a USB 2.0 cable (and the test build linked in #164) gets me a clean rip.

Using a USB 3.0 gets me an error at the end of the rip - `IoctlFailed` similar to the original reporter.

Like the original reporter, EAC and this drive play nice together with USB 2 and USB 3.
